dmt/A_mappers/ros_A_mapper.py

```
@@ -0,0 +1,1024 @@
+        f.write(spacing + '{\n')
+        if self.size is None:
+            # Use length
+            f.write(spacing + '  uint32_t length = %s_length;\n' % self.name)
+            f.write(spacing + '  var->%s.nCount = length;\n' % self.name)
+        else:
+            # Copy all
+            f.write(spacing + '  uint32_t length = %d;\n' % self.size)
+        f.write(spacing + '  memcpy(var->%s, %s, length * sizeof(*%s));\n' %
+                (self.name, self.name, self.name))
+        f.write(spacing + '}\n')
+
+
+
+# Messages
+
+class Message:
+    """ Parses message definitions into something we can export. """
+    global ROS_TO_EMBEDDED_TYPES
+
+    def __init__(self, name: str, package: str, definition: str, md5: str) -> None:
+
+        self.name = name            # name of message/class
+        self.package = package      # package we reside in
+        self.md5 = md5              # checksum
+        self.includes = list()      # other files we must include
+
+        self.data = list()          # data types for code generation
+        self.enums = list()
+
+        self.message_text = definition
+
+        # parse definition
+        definition = definition.splitlines()
+        for line in definition:
+            # prep work
+            line = line.strip().rstrip()
+            value = None
+            # Remove the comments
+            if line.find("#") > -1:
+                line = line[0:line.find("#")]
+
+            # Get the value of a variable if present
+            if line.find("=") > -1:
+                try:
+                    value = line[line.find("=")+1:]
+                except Exception as e:
+                    value = '"' + line[line.find("=")+1:] + '"'
+                line = line[0:line.find("=")]
+
+            # find package/class name
+            line = line.replace("\t", " ")
+            l = line.split(" ")
+            while "" in l:
+                l.remove("")
+            if len(l) < 2:
```

`len(l) < 2` and `len(l) > 2` are errors? Prefer not to silently ignore errors. at least print a message.

---

Yes it should be just a type and a variable name. That's why it is checking and also removing any empty elements before. I can print a message indeed.